docs/OIDC-Provider-Examples/sap-public-oidc-provider.md

```diff
@@ -4,33 +4,54 @@ SAP IAS can be setup as an [OIDC provider](https://help.sap.com/viewer/6d6d63354
+   * E.g. for a UAA part of a CF-Deployment, this is `https://login.cf.<domain>/login/callback/{origin}`
+   * `{origin}` - is the id you of the OIDC provider you will use in UAA in the next step
    
-2. Copy client id.
+2. Go to "Client Authentication" section and check "Allow Public Client Flows".
+   * This will generate the "client id" on the top of the page
+   * Copy "client id", to use for the uaa configuration.
 
-3. Minimal OIDC configuration needs to be added in login.ym.
-   Read configuration refer to '[https://<tenant ID>.accounts.ondemand.com/.well-known/openid-configuration](https://help.sap.com/viewer/6d6d63354d1242d185ab4830fc04feb1/Cloud/en-US/c297516bae4547eb82eeed80fea2b937.html)' for discoveryUrl and issuer
+4. Minimal OIDC configuration needs to be added in `uaa.yml` or `login.yml` (depending on the setup).
+   Read configuration refer to '[https://<tenant ID>.accounts.ondemand.com/.well-known/openid-configuration](https://help.sap.com/viewer/6d6d63354d1242d185ab4830fc04feb1/Cloud/en-US/c297516bae4547eb82eeed80fea2b937.html)' for discoveryUrl and issuer. E.g. in the example below `ias.public` was selected as `{origin}`
 
         login:
           oauth:
             providers:
               ias.public:
                 type: oidc1.0
-                discoveryUrl: https://trailaccount.accounts.ondemand.com/.well-known/openid-configuration
-                issuer: https://trailaccount.accounts.ondemand.com
+                discoveryUrl: https://<ias_tenant_id>.accounts.ondemand.com/.well-known/openid-configuration
+                issuer: https://<ias_tenant_id>.accounts.ondemand.com
                 scopes:
                   - openid
                   - email
                   - profile
                 linkText: Login with IAS-Public
                 showLinkText: true
-                relyingPartyId: 3feb7ecb-d106-4432-b335-aca2689ad123
+                relyingPartyId: <client_id>
+                addShadowUserOnLogin: true
+
+6. Ensure that the scope `openid`, `email` and `profile` is included in the `scopes` property. Then UAA shadow user (if addShadowUserOnLogin=true) is created with all properties. 
 
-4. Ensure that the scope `openid`, `email` and `profile` is included in the`scopes` property. Then UAA shadow user (if addShadowUserOnLogin=true) is created
-   with all properties. 
+7. Restart UAA.
+   * You may see `Login with IAS-Public` link on your login page.
+   * Or if the link is not displayed, you need to enter the `{origin}` manually and then login against it
 
-5. Restart UAA. You will see `Login with IAS-Public` link on your login page.
+9. (optional) For CF Login, use `cf login --sso` and select the provider.
+    * Trying to login with User/Pass requires a confidential OAuth Client, creating a Secret in the Client Authentication tab, adding it as `relyingPartySecret` property and disabling "Enforce PKCE"
```

Comment:
UAA by default uses now always PKCE

Reply:
@strehle , then in point 1., do we need to explicitly state that the new application should be made "public", because this confused me a bit, since there was no place to mark "public".
Maybe we could say there, that the application is by default "public", and a confidential one is required, a secret should be added? Or how the application can be made not public ?